employees/tests/test_views.py

```diff
@@ -939,3 +838,157 @@ def _get_total_hours_per_project_and_percentage_from_month(self):
+        self.assertEqual(
+            response.context_data["form"].errors["task_activities"][0],
+            ReportValidationStrings.TASK_ACTIVITY_NOT_RELATED_TO_PROJECT.value,
+        )
+        self.assertCountEqual(author_reports_before_post, self.user.report_set.all())
+
+    def test_create_report_form_view_should_contain_only_active_projects_related_to_user(self):
+        other_project = ProjectFactory()
+
+        response = self.client.get(self.url)
+        project_choices = response.context["form"].fields["project"].queryset
+
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(self.report.project, project_choices)
+        self.assertNotIn(other_project, project_choices)
+        self.assertIn(self.report.project, Project.objects.filter_active())
+
+    def test_create_report_form_should_have_empty_project_and_activities_fields_if_user_is_not_in_any_active_project(
+        self
+    ):
+        self.user.projects.remove(self.report.project)
+
+        response = self.client.get(self.url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.context["form"].fields["project"].queryset), 0)
+        self.assertEqual(len(response.context["form"].fields["task_activities"].queryset), 0)
+
+    def test_create_report_form_projects_field_for_manager_should_contain_only_projects_where_he_is_manager(self):
+        manager = ManagerUserFactory()
+        self.client.force_login(manager)
+        self.report.project.managers.add(manager)
+
+        response = self.client.get(self.url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(self.report.project, response.context["form"].fields["project"].queryset)
+
+
+class JoinProjectViewTests(TestCase):
+    def setUp(self):
+        self.user = UserFactory()
+        self.client.force_login(self.user)
+        self.task_activity = TaskActivityTypeFactory(is_default=True)
+        self.report = ReportFactory(
+            author=self.user,
+            date=datetime.datetime.now().date(),
+            task_activities=self.task_activity,
+            work_hours=datetime.timedelta(hours=8),
+        )
+        self.report.project.members.add(self.user)
+        self.url = reverse(
+            "join-project",
+            kwargs={"year": datetime.datetime.now().date().year, "month": datetime.datetime.now().date().month},
+        )
+        self.success_url = reverse(
+            "custom-report-list",
+            kwargs={"year": datetime.datetime.now().date().year, "month": datetime.datetime.now().date().month},
+        )
+
+    def test_join_project_view_should_add_user_to_project_selected_in_project_join_form_on_post(self):
+        new_project = ProjectFactory(name="New Project", start_date=datetime.datetime.now())
+        response = self.client.post(path=self.url, data={"projects": new_project.id})
+        self.assertEqual(response.status_code, 302)
```

Comment:
It should be 201 or 200 :)

Reply:
Again - I believe that this is proper behavior in this case.